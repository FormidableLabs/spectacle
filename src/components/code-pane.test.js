import React from 'react';
import { mount } from 'enzyme';
import CodePane from './code-pane';

describe('<CodePane />', () => {
  test('should render correctly.', () => {
    const context = {
      styles: {
        components: { codePane: {}, syntax: {} },
        prism: { light: 'light;', dark: 'dark;' }
      }
    };
    const source = `
      const myButton = (
        <CustomButton
          style={{ background: '#f00' }}
          onClick={this.action}
        >
         Click Me
        </CustomButton>
      );
    `;
    const wrapper = mount(<CodePane lang="jsx" source={source} />, { context });
    expect(wrapper).toMatchSnapshot();
  });
});
