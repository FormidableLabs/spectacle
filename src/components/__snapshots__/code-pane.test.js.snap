// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodePane /> should render correctly. 1`] = `
<CodePane
  lang="jsx"
  source="
      const myButton = (
        <CustomButton
          style={{ background: '#f00' }}
          onClick={this.action}
        >
         Click Me
        </CustomButton>
      );
    "
>
  <pre
    data-radium={true}
    style={Object {}}
  >
    <code
      className="language-jsx"
      dangerouslySetInnerHTML={
        Object {
          "__html": "
              const myButton = (
                &lt;CustomButton
                  style={{ background: '#f00' }}
                  onClick={this.action}
                &gt;
                 Click Me
                &lt;/CustomButton&gt;
              );
            ",
        }
      }
    />
  </pre>
</CodePane>
`;
